include/unifex/await_transform.hpp

```diff
@@ -73,7 +74,12 @@ class _as_awaitable {
     
     void set_done() && noexcept {
       op_->state_ = state::done;
-      continuation_.resume();
+      if constexpr (remove_cvref_t<Sender>::sends_done) {
+        continuation_.promise().unhandled_done().resume();
+      } else {
+        // should be unreachable
+        std::terminate();
```

Comment:
If this is *actually* unreachable, aren't you converting a compile-time check into a runtime crash with this?  Why not lean on a `static_assert` instead?

Reply:
Yes, I had meant to come back and replace this with use of a `UNIFEX_UNREACHABLE()` macro, but a `static_assert` would be better. I'm just not sure if we can guarantee this method will never be instantiated in all cases, even if it's not called. I guess at least we'd know if it did end up getting instantiated :)

Reply:
I've updated to a static_assert and it seems to still compile!